src/js/node/tls.ts

```diff
@@ -308,14 +308,19 @@ function TLSSocket(socket?, options?) {
   this[krenegotiationDisabled] = undefined;
   this.encrypted = true;
 
-  NetSocket.$call(this, socket instanceof NetSocket || socket instanceof Duplex ? options : options || socket);
-  options = options || socket || {};
+  const isNetSocketOrDuplex = socket instanceof NetSocket || socket instanceof Duplex;
```

Comment:
`NetSocket` extends `Duplex`, so in theory we should only need to check instanceof Duplex:

```suggestion
  const isNetSocketOrDuplex = socket instanceof Duplex;
```